common/djangoapps/entitlements/models.py

```diff
@@ -302,6 +304,45 @@ def get_active_entitlements_for_user(cls, user):
             enrollment_course_run=None
         ).select_related('user').select_related('enrollment_course_run')
 
+    @classmethod
+    def check_for_existing_entitlement_and_enroll(cls, user, course_run_key):
+        """
+        Looks at the User's existing entitlements to see if the user already has a Course Entitlement for the course run
+        provided in the course_key.  If the user does have an Entitlement with no run set, the User is enrolled in the mode
+        set in the Entitlement.
+
+        Note: This method assumes that eligibility to enroll in the course has already been checked.
+
+        Arguments:
+            user (User): The user that we are inspecting the entitlements for.
+            course_run_key (CourseKey): The course run Key.
+
+        Returns:
+            CourseEntitlement: The CourseEntitlement that was found and used and None otherwise.
+        """
+        # Check if the User has any Entitlements that are active
+        entitlements = CourseEntitlement.get_active_entitlements_for_user(user)
+        if len(entitlements) > 0:
+            course_uuid_str = get_course_uuid_for_course(course_run_key)
```

Comment:
again the _for_course piece of this function makes me wonder if this should be a Course function? I forget which model is the course level model in edx-platform, but something like Course.get_course_uuid should probably be available right?